src/cart/crt_iv.c

```diff
@@ -1704,18 +1703,20 @@ crt_hdlr_iv_sync_aux(void *arg)
 	case CRT_IV_SYNC_EVENT_UPDATE:
 	{
 		d_sg_list_t tmp_iv;
+		d_iov_t	    tmp_iov = { 0 };
 
 		rc = iv_ops->ivo_on_get(ivns_internal, &input->ivs_key,
-				0, CRT_IV_PERM_WRITE, &iv_value, &user_priv);
+				0, CRT_IV_PERM_WRITE, NULL, &user_priv);
 
-		tmp_iv = iv_value;
 		if (rc != 0) {
 			D_ERROR("ivo_on_get() failed; rc=%d\n", rc);
 			D_GOTO(exit, rc);
 		}
 
 		need_put = true;
 
+		tmp_iv.sg_nr = 1; /* Only use single bulk seg for IV for now */
```

Comment:
I don't like this hardcoding of 1; the whole reason for having ivo_on_get() is to remove any knowledge of structure of iv value from cart internals -- instead iv framework should call ivo_on_get() which would provide back 'structure' of the iv value.

As such, instead the line 1709 should be instead to CRT_IV_PERM_READ, for which the implementation of ivo_on_get() should return iv_value, and based on it we should populate tmp_iv.sg_nr here.

The same comment applies to all other places with similar change to this one

Reply:
Hmm, the issue is where to free the buffer allocated inside ivo_on_get().  So even using using CRT_IV_PERM_READ, and then on_get only allocate daos_iov_t, (not the real buffer), 
but we will still need free the iov_buf. 

In these 2 cases, those iov_buffer is only used to retrieve the buffer from the bulk, it does not really need DAOS to tell buffer size etc.

Reply:
the buffer should be freed in on_put(). am not sure i understand why it was not working, since iv_value is not being modified previously -- we only copy contents to tmp_iv, so when we call on_put, it should get invoked with whatever data was provided during on_get() inside of iv_value

Reply:
Discussed with Alex, he prefer to fix this in a different way.